app/src/main/java/ch/epfl/sdp/map/QuadrilateralBuilder.kt

```
@@ -0,0 +1,52 @@
+package ch.epfl.sdp.map
+
+import ch.epfl.sdp.searchareabuilder.QuadrilateralArea
+import ch.epfl.sdp.searchareabuilder.SearchArea
+import ch.epfl.sdp.searchareabuilder.SearchAreaNotCompleteException
+import ch.epfl.sdp.ui.maps.IntersectionTools
+import com.mapbox.mapboxsdk.geometry.LatLng
+import java.util.*
+
+class QuadrilateralBuilder : SearchAreaBuilder() {
+
+    override fun addVertex(vertex: LatLng) {
+        if (vertices.size >= 4) return

```

Should we silently return or throw an exception ?

---

Your right, we should consider return a boolean. Or even an exception might also be a solution and we could consider displaing it's message via a toast on screen during the build process.

---

Let's go with the exception and make the method return `this` to enable chained method calls

---

Done